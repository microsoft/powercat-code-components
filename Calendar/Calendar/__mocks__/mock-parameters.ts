/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockDateTimeProperty, MockEnumProperty, MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        SelectedDateValue: new MockDateTimeProperty(),
        MinDate: new MockDateTimeProperty(),
        MaxDate: new MockDateTimeProperty(),
        ShowGoToToday: new MockTwoOptionsProperty(),
        MonthPickerVisible: new MockTwoOptionsProperty(),
        DayPickerVisible: new MockTwoOptionsProperty(),
        HighlightSelectedMonth: new MockTwoOptionsProperty(),
        HighlightCurrentMonth: new MockTwoOptionsProperty(),
        ShowSixWeeksByDefault: new MockTwoOptionsProperty(),
        ShowWeekNumbers: new MockTwoOptionsProperty(),
        FirstDayOfWeek: new MockEnumProperty(),
        BackgroundColor: new MockStringProperty(),
    };
}
