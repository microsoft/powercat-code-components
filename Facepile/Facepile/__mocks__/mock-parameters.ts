/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayablePersonas: new MockWholeNumberProperty(),
        ImageShouldFadeIn: new MockTwoOptionsProperty(),
        PersonaSize: new MockEnumProperty(),
        OverflowButtonType: new MockEnumProperty(),
        ShowAddButton: new MockTwoOptionsProperty(),
        OverflowButtonAriaLabel: new MockStringProperty(),
        AddbuttonAriaLabel: new MockStringProperty(),
    };
}
