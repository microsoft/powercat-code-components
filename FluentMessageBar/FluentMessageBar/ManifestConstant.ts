export const enum ItemColumns {
  DisplayName = 'ItemDisplayName',
  Key = 'ItemKey',
  IconName = 'ItemIconName',
  IconStyle = 'ItemIconStyle',
  Visible = 'ItemVisible',
  Disabled = 'ItemDisabled',
  Appearance = 'ItemAppearance',
  Tooltip = 'ItemTooltip',
}

export const enum StringConstants {
  Regular = 'Regular',
  Subtle = 'subtle',
  AboveHeader = 'Above header',
  BelowHeader = 'Below header',
}

export const Size = {
    Large: 'large',
    Medium: 'medium',
    Small: 'small',
};

export const Layout = {
    'Icon only': 'icon',
    'Text only': 'text',
    'Icon before': 'before',
    'Icon after': 'after',
    'Icon above': 'above',
};

export const enum ManifestPropertyNames {
  dataset = 'dataset',
}

export const Orientation = {
    Vertical: 'vertical',
    Horizontal: 'horizontal',
};

export const enum Alignment {
  Alignment = 'Alignment',
}
