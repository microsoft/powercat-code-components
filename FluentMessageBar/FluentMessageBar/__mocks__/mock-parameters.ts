/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty } from "./mock-context";
import { MockDataSet } from "./mock-datasets";

export function getMockParameters(): IInputs {
  return {
    Title: new MockStringProperty(),
    Items: new MockDataSet([]),
    Body:new MockStringProperty(),
    LinkText:new MockStringProperty(),
    URL:new MockStringProperty(),
    HideDismiss:new MockTwoOptionsProperty(),
    Intent: new MockEnumProperty(),
    Shape:new MockEnumProperty()


  };
}
