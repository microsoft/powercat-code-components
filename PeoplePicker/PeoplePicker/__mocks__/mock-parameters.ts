/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Personas: new MockDataSet([]),
        Suggestions: new MockDataSet([]),
        Error: new MockTwoOptionsProperty(),
        ShowSecondaryText: new MockTwoOptionsProperty(),
        PeoplePickerType: new MockEnumProperty(),
        MinimumSearchTermLength: new MockWholeNumberProperty(),
        SearchTermToShortMessage: new MockStringProperty(),
        SuggestionsHeaderText: new MockStringProperty(),
        NoResultFoundMessage: new MockStringProperty(),
        HintText: new MockStringProperty(),
        MaxPeople: new MockWholeNumberProperty(),
    };
}
