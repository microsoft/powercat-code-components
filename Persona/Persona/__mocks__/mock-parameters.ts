/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        PersonaSize: new MockEnumProperty(),
        ImageUrl: new MockStringProperty(),
        ImageAlt: new MockStringProperty(),
        Presence: new MockEnumProperty(),
        ImageInitials: new MockStringProperty(),
        OptionalText: new MockStringProperty(),
        HidePersonaDetails: new MockTwoOptionsProperty(),
        TertiaryText: new MockStringProperty(),
        Text: new MockStringProperty(),
        SecondaryText: new MockStringProperty(),
        PersonaInitialsColor: new MockEnumProperty(),
    };
}
