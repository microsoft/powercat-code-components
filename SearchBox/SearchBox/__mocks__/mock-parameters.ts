/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        Underlined: new MockTwoOptionsProperty(),
        PlaceHolderText: new MockStringProperty(),
        IconName: new MockStringProperty(),
        DisableAnimation: new MockTwoOptionsProperty(),
        InputEvent: new MockStringProperty(),
    };
}
