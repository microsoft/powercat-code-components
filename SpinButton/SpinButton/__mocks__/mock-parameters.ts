/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Min: new MockWholeNumberProperty(),
        Max: new MockWholeNumberProperty(),
        Step: new MockWholeNumberProperty(),
        Label: new MockStringProperty(),
        IconName: new MockStringProperty(),
        DefaultValue: new MockStringProperty(),
        Value: new MockStringProperty(),
        Suffix: new MockStringProperty(),
        DelayOutput: new MockTwoOptionsProperty(),
        Error: new MockTwoOptionsProperty(),
        InputEvent: new MockStringProperty(),
    };
}
