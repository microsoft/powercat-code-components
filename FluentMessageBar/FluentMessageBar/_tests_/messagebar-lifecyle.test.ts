
import { FluentMessageBar } from '..';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { MockDataSet, MockEntityRecord } from '../__mocks__/mock-datasets';
import { getMockParameters } from '../__mocks__/mock-parameters';
import { IInputs, IOutputs } from '../generated/ManifestTypes';
import { ItemColumns } from '../ManifestConstant';

describe('FluentMessageBar', () => {
    let component: FluentMessageBar;
    let context: ComponentFramework.Context<IInputs>;
    let notifyOutputChanged: jest.Mock;
    let container: HTMLDivElement;

    beforeEach(() => {
        notifyOutputChanged = jest.fn();
        container = document.createElement('div');
        component = new FluentMessageBar();
        jest.clearAllMocks();
    });
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === "DIV") {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
    });


    it("renders", () => {
        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged, {}, container);
        const messageBarElement = component.updateView(context);
        expect(messageBarElement).toMatchSnapshot();
    });

    it('should update view correctly', async () => {

        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged, {}, container);
        const messageBarElement = component.updateView(context);
        expect(messageBarElement).toMatchSnapshot();

        expect(component['_context']).toBe(context);
        expect(component['notifyOutputChanged']).toBe(notifyOutputChanged);
        expect(component['parentContainer']).toBe(container);

    });

    it('should call OnDismiss event', async () => {
        context = {
            events: {
                OnDismiss: jest.fn()
            },

            mode: {
                trackContainerResize: jest.fn(),
                allocatedHeight: 0,
                allocatedWidth: 0,
                isControlDisabled: false,
                isVisible: false,
                label: ''
            }

        } as unknown as ComponentFramework.Context<IInputs>;
        // Act: Call init and let updateView proceed without mocking
        await component.init(context, notifyOutputChanged, {}, container);
        component.OnDismiss();
        expect(context.events.OnDismiss).toHaveBeenCalled();
    });

    it('should return empty outputs', () => {
        const outputs: IOutputs = component.getOutputs();
        expect(outputs).toEqual({});
    });
    
    it('should return default value if property is empty', () => {
        const defaultValue = 'default';
        const result = component.defaultIfEmpty({
            raw: null,
            error: false,
            errorMessage: '',
            type: ''
        }, defaultValue);
        expect(result).toBe(defaultValue);
    });

    it('should return property value if not empty', () => {
        const value = 'value';
        const result = component.defaultIfEmpty({
            raw: value,
            error: false,
            errorMessage: '',
            type: ''
        }, 'default');
        expect(result).toBe(value);
    });

});


function createComponent() {
    const component = new FluentMessageBar();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());
    context.parameters.Title.raw = "Title";
    context.parameters.Body.raw = "Body";
    //context.mode.isVisible = true;
    context.parameters.Shape.raw = "square";
    context.parameters.LinkText.raw = "Link";
    context.parameters.URL.raw = "https://google.com";
    context.parameters.Intent.raw = 'info';
    context.parameters.Items = new MockDataSet([
        new MockEntityRecord("1", {
            [ItemColumns.Key]: "toolAdd",
            [ItemColumns.DisplayName]: "Add",
            [ItemColumns.IconName]: "Add",
            [ItemColumns.IconStyle]: "Filled",
            [ItemColumns.Appearance]: "Primary",
            [ItemColumns.Visible]: true,
        }),
        new MockEntityRecord("2", {
            [ItemColumns.Key]: "toolEdit",
            [ItemColumns.DisplayName]: "Edit",
            [ItemColumns.IconName]: "Edit",
            [ItemColumns.IconStyle]: "Regular",
            [ItemColumns.Appearance]: "Subtle",
            [ItemColumns.Visible]: true,
        }),
        new MockEntityRecord("3", {
            [ItemColumns.Key]: "toolDelete",
            [ItemColumns.DisplayName]: "Delete",
            [ItemColumns.IconName]: "Delete",
            [ItemColumns.Disabled]: true,
            [ItemColumns.IconStyle]: "Filled",
            [ItemColumns.Appearance]: "Subtle",
            [ItemColumns.Visible]: true,
        }),
        new MockEntityRecord("4", {
            [ItemColumns.Key]: "toolInfo",
            [ItemColumns.DisplayName]: "Info",
            [ItemColumns.IconName]: "Info",
            [ItemColumns.Disabled]: false,
            [ItemColumns.IconStyle]: "Filled",
            [ItemColumns.Appearance]: "Subtle",
            [ItemColumns.Visible]: false,
        }),
    ]);

    const state = new MockState();
    const container = document.createElement("div");
    document.body.appendChild(container);
    return { component, context, container, notifyOutputChanged, state };
}