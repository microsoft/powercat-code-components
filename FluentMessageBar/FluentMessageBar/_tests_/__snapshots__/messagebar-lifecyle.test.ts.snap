// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FluentMessageBar renders 1`] = `
<MessageBarComponent
  OnDismiss={[Function]}
  Url="https://google.com"
  body="Body"
  getPopoverRoot={[Function]}
  height={-1}
  items={
    [
      {
        "appearance": "primary",
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "1",
          "values": {
            "ItemAppearance": "Primary",
            "ItemDisplayName": "Add",
            "ItemIconName": "Add",
            "ItemIconStyle": "Filled",
            "ItemKey": "toolAdd",
            "ItemVisible": true,
          },
        },
        "disabled": false,
        "iconName": "Add",
        "iconStyle": "Filled",
        "id": "1",
        "key": "toolAdd",
        "name": "Add",
        "tooltip": undefined,
        "visible": true,
      },
      {
        "appearance": "subtle",
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "2",
          "values": {
            "ItemAppearance": "Subtle",
            "ItemDisplayName": "Edit",
            "ItemIconName": "Edit",
            "ItemIconStyle": "Regular",
            "ItemKey": "toolEdit",
            "ItemVisible": true,
          },
        },
        "disabled": false,
        "iconName": "Edit",
        "iconStyle": "Regular",
        "id": "2",
        "key": "toolEdit",
        "name": "Edit",
        "tooltip": undefined,
        "visible": true,
      },
    ]
  }
  linkText="Link"
  messageBarIntent="info"
  messageBarShape="square"
  onResize={[Function]}
  onSelected={[Function]}
  title="Title"
  width={-1}
/>
`;

exports[`FluentMessageBar should update view correctly 1`] = `
<MessageBarComponent
  OnDismiss={[Function]}
  Url="https://google.com"
  body="Body"
  getPopoverRoot={[Function]}
  height={-1}
  items={
    [
      {
        "appearance": "primary",
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "1",
          "values": {
            "ItemAppearance": "Primary",
            "ItemDisplayName": "Add",
            "ItemIconName": "Add",
            "ItemIconStyle": "Filled",
            "ItemKey": "toolAdd",
            "ItemVisible": true,
          },
        },
        "disabled": false,
        "iconName": "Add",
        "iconStyle": "Filled",
        "id": "1",
        "key": "toolAdd",
        "name": "Add",
        "tooltip": undefined,
        "visible": true,
      },
      {
        "appearance": "subtle",
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "2",
          "values": {
            "ItemAppearance": "Subtle",
            "ItemDisplayName": "Edit",
            "ItemIconName": "Edit",
            "ItemIconStyle": "Regular",
            "ItemKey": "toolEdit",
            "ItemVisible": true,
          },
        },
        "disabled": false,
        "iconName": "Edit",
        "iconStyle": "Regular",
        "id": "2",
        "key": "toolEdit",
        "name": "Edit",
        "tooltip": undefined,
        "visible": true,
      },
    ]
  }
  linkText="Link"
  messageBarIntent="info"
  messageBarShape="square"
  onResize={[Function]}
  onSelected={[Function]}
  title="Title"
  width={-1}
/>
`;
